# **(hypothesis testing)**

## Objectives

* Write your notebook objective here, for example, "Fetch data from Kaggle and save as raw data", or "engineer features for modelling"

## Inputs

* Write down which data or information you need to run the notebook 
* Write here which files, code or artefacts you generate by the end of the notebook 

## Additional Comments

* If you have any additional comments that don't fit in the previous bullets, please state them here. 



* Loading of the dataset as Pandas DataFrame.

In [1]:
import pandas as pd
import numpy as np

* 'pd.read_csv()'

In [2]:
import pandas as pd
df_cleaned = pd.read_csv('../jupyter_notebooks/final/cleaned_shopping_trends_and_customer_behaviour_dataset.csv') 
df_cleaned

age       age_group  gender item_purchased     category  \
0      55  Pre-Retirement    Male         Blouse     Clothing   
1      19           Gen Z    Male        Sweater     Clothing   
2      50      Gen X Late    Male          Jeans     Clothing   
3      21           Gen Z    Male        Sandals     Footwear   
4      45      Gen X Late    Male         Blouse     Clothing   
...   ...             ...     ...            ...          ...   
3895   40     Gen X Early  Female         Hoodie     Clothing   
3896   52      Gen X Late  Female       Backpack  Accessories   
3897   46      Gen X Late  Female           Belt  Accessories   
3898   44     Gen X Early  Female          Shoes     Footwear   
3899   52      Gen X Late  Female        Handbag  Accessories   

      purchase_amount_(usd)       location     colour  season  review_rating  \
0                        53       Kentucky       Gray  Winter            3.1   
1                        64          Maine     Maroon  Winter            3.1   
2                        73  Massachusetts     Maroon  Spring            3.1   
3                        90   Rhode Island     Maroon  Spring            3.5   
4                        49         Oregon  Turquoise  Spring            2.7   
...                     ...            ...        ...     ...            ...   
3895                     28       Virginia  Turquoise  Summer            4.2   
3896                     49           Iowa      White  Spring            4.5   
3897                     33     New Jersey      Green  Spring            2.9   
3898                     77      Minnesota      Brown  Summer            3.8   
3899                     81     California      Beige  Spring            3.1   

     subscription_status   shipping_type discount_applied promo_code_used  \
0                    Yes         Express              Yes             Yes   
1                    Yes         Express              Yes             Yes   
2                    Yes   Free Shipping              Yes             Yes   
3                    Yes    Next Day Air              Yes             Yes   
4                    Yes   Free Shipping              Yes             Yes   
...                  ...             ...              ...             ...   
3895                  No  2-Day Shipping               No              No   
3896                  No    Store Pickup               No              No   
3897                  No        Standard               No              No   
3898                  No         Express               No              No   
3899                  No    Store Pickup               No              No   

      previous_purchases payment_method frequency_of_purchases  
0                     14          Venmo              Bi-Weekly  
1                      2           Cash              Bi-Weekly  
2                     23    Credit Card                 Weekly  
3                     49         PayPal                 Weekly  
4                     31         PayPal               Annually  
...                  ...            ...                    ...  
3895                  32          Venmo                 Weekly  
3896                  41  Bank Transfer              Bi-Weekly  
3897                  24          Venmo              Quarterly  
3898                  24          Venmo                 Weekly  
3899                  33          Venmo              Quarterly  

[3900 rows x 17 columns]

In [3]:
import scipy.stats as stats

* H1: Younger customer groups use promo codes more often than older groups.

In [ ]:
# Clean and preprocess columns
df_cleaned['promo_code_used'] = df_cleaned['promo_code_used'].str.strip().str.lower()

# H1: Age group vs Promo code usage (Chi-squared)
h1_table = pd.crosstab(df_cleaned['age_group'], df_cleaned['promo_code_used'])
h1_chi2, h1_p, _, _ = stats.chi2_contingency(h1_table)

# Display results
print("H1 - Age Group vs Promo Code Usage:")
print("Chi² Stat:", h1_chi2)
print("p-value:", h1_p)
print("Significant:", h1_p < 0.05)
print("\n")

H1 - Age Group vs Promo Code Usage:
Chi² Stat: 1.0887155051993171
p-value: 0.9550976431155631
Significant: False




* Interpretation: The p-value is much greater than 0.05, so we fail to reject the null hypothesis. There is no statistically significant difference in promo code usage across age groups.

* H2: Female customers are more likely to use discounts than male customers.

In [ ]:
# Clean and preprocess columns
df_cleaned['discount_applied'] = df_cleaned['discount_applied'].str.strip().str.lower()

# H2: Gender vs Discount usage (Chi-squared)
h2_table = pd.crosstab(df_cleaned['gender'], df_cleaned['discount_applied'])
h2_chi2, h2_p, _, _ = stats.chi2_contingency(h2_table)

# Display results
print("H2 - Gender vs Discount Usage:")
print("Chi² Stat:", h2_chi2)
print("p-value:", h2_p)
print("Significant:", h2_p < 0.05)
print("\n")

H2 - Gender vs Discount Usage:
Chi² Stat: 1381.9413463923058
p-value: 1.7645380691667684e-302
Significant: True




* Interpretation: The p-value is far below 0.05 [1.76e−302 = in plain terms to 0.000...(301 zeros)...176] so we reject the null hypothesis. There is a strong correlation between gender and discount usage. Female customers are significantly more likely to use discounts than male customers.

* H3: Customers who use discounts tend to spend more per purchase.

In [ ]:
# Clean and preprocess columns
df_cleaned['gender'] = df_cleaned['gender'].str.strip().str.capitalize()

# H3: Spend difference by discount usage (t-test)
group1 = df_cleaned[df_cleaned['discount_applied'] == 'yes']['purchase_amount_(usd)']
group2 = df_cleaned[df_cleaned['discount_applied'] == 'no']['purchase_amount_(usd)']
h3_t, h3_p = stats.ttest_ind(group1, group2, equal_var=False)

# Display results
print("H3 - Discount vs Spend:")
print("Chi² Stat:", h3_p)
print("p-value:", h3_t)
print("Significant:", h3_p < 0.05)
print("\n")

H3 - Discount vs Spend:
Chi² Stat: 0.26611244488939256
p-value: -1.1122316185807992
Significant: False




* Interpretation: The p-value is greater than 0.05, so we fail to reject the null hypothesis. There is no significant difference in spending between customers who use discounts and those who do not.

* H4: Season affects promotion response (e.g., more promo code use during Winter sales)

In [12]:
# Clean and preprocess columns
df_cleaned['age_group'] = df_cleaned['age_group'].str.strip()

# H4: Season vs Promo code usage (Chi-squared)
h4_table = pd.crosstab(df_cleaned['season'], df_cleaned['promo_code_used'])
h4_chi2, h4_p, _, _ = stats.chi2_contingency(h4_table)

# Display results
print("H4 - Season vs Promo Code Usage:")
print("Chi² Stat:", h4_chi2)
print("p-value:", h4_p)
print("Significant:", h4_p < 0.05)
print("\n")

H4 - Season vs Promo Code Usage:
Chi² Stat: 3.1687022338896487
p-value: 0.3663388240993125
Significant: False




* Interpretation: The p-value is greater than 0.05, so we fail to reject the null hypothesis. There is no statistically significant correlation between season and promo code usage.